src/D2L.CodeStyle.Analyzers/Language/RequireNamedArgumentsAnalyzer.cs

```diff
@@ -65,7 +65,24 @@ SyntaxNodeAnalysisContext ctx
 				return;
 			}
 
-			var unnamedArgs = GetUnnamedArgs(
+            // Expression trees aren't compatible with named arguments,
+            // so skip any expressions
+            var implicitType = ctx.SemanticModel.GetTypeInfo(ctx.Node.Parent).ConvertedType;
+            if (implicitType != null) {
+                // Only lambda type expressions have arguments,
+                // so this only applies to LambdaExpression
+                var expressionType = ctx.SemanticModel.Compilation.GetTypeByMetadataName("System.Linq.Expressions.LambdaExpression");
+                var baseExprType = implicitType.BaseType;

```

Comment:
Uhm, double check that this doesn't cause errors: `new object( new object( new object( ) ) );`

Reply:
No issues with that